clippy_lints/src/methods.rs

```diff
@@ -329,6 +329,32 @@ declare_clippy_lint! {
     "using any `*or` method with a function call, which suggests `*or_else`"
 }
 
+/// **What it does:** Checks for calls to `.expect(&format!(...))`, `.expect(foo(..))`,
+/// etc., and suggests to use `unwrap_or_else` instead
+///
+/// **Why is this bad?** The function will always be called.
+///
+/// **Known problems:** If the function has side-effects, not calling it will
+/// change the semantic of the program, but you shouldn't rely on that anyway.
+/// 
+/// **Example:**
+/// ```rust
+/// foo.expect(&format("Err {}: {}", err_code, err_msg))
+/// ```
+/// or
+/// ```rust
+/// foo.expect(format("Err {}: {}", err_code, err_msg).as_str())
+/// ```
+/// this can instead be written:
+/// ```rust
+/// foo.unwrap_or_else(|_| panic!(&format("Err {}: {}", err_code, err_msg)))
```

Comment:
The docs still have the old example

Reply:
Bah, my bad, I knew I'd forget something. Hopefully fixed now.